### DISCUSSION TITLE
Why do Tags that I create, always have "assets"

### DISCUSSION THREAD

**Discussion by The-Futurist** [community/community#52189] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Repositories

### Body

I've hardly ever used tags, but created one eariler and pushed that to a fork. In GitHub I can see that Tags have assets and these seem to just a zip of the entire repo!

Why is that? I thought a tag was more or less just like a permanent branch, just a pointer to a commit so why is the entire (huge) repo zipped up and saved like this?

**Comment by AneeqMalik** [comment]
Can you plz be precise which tags are you refering to?

**Reply by The-Futurist** [reply]
Of course @AneeqMalik, take look at this:

![image](https://user-images.githubusercontent.com/12262952/230507273-e161de76-6f5c-4f4e-8f25-a4c5b802ede6.png)

Selecting the first:

![image](https://user-images.githubusercontent.com/12262952/230507359-5816dd2a-967a-4b8e-9ba6-8ebe8cfba376.png)

First it would be helpful to include the size of these downloadable assets and second, what the heck are they? why are they there and why can I find no GitHub documentation that explains anything about them? why are they named "Source code"? which source code? the whole repository??

**Comment by airtower-luna** [comment]
It's just the content of the repository *as of that tag*, for anyone who finds downloading that more convenient that cloning the repository.

At least as of now those files aren't permanently saved but rather generated when someone downloads them. That's evident because a [recent change](https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/) changed their checksums (because the archive generation method changed), and then the change was rolled back because the different checksums caused trouble for a lot of people, see #45830. :sweat_smile:

**Reply by The-Futurist** [reply]
@airtower-luna - OK that's helpful to understand, much appreciated.
